snap/component.go

```diff
@@ -160,8 +160,8 @@ func (c *componentPlaceInfo) MountDescription() string {
 
 // ComponentLinkPath returns the path for the symlink for a component for a
 // given snap revision.
-func ComponentLinkPath(cpi ContainerPlaceInfo, snapRev Revision) string {
-	instanceName, compName, _ := strings.Cut(cpi.ContainerName(), "+")
+func ComponentLinkPath(cref naming.ComponentRef, snapRev Revision) string {
+	instanceName, compName, _ := strings.Cut(cref.String(), "+")
```

Comment:
do we need the Cut? we could use cref.SnapName and cref.ComponentName, no?

Reply:
Whoops, that was part of the motivation for this change.

Reply:
That's correct, it should carry the snap name, not the instance name in principle.